swap_meet/vendor.py

```diff
@@ -1,2 +1,61 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        cateogry_list = []
+        for item in self.inventory:
+            if item.category == category :
+                cateogry_list.append(item)                
+        return cateogry_list
+
+    def swap_items(self, friend, item_give, item_get):
+        if item_give in self.inventory and item_get in friend.inventory:
+            friend.inventory.append(item_give)
+            self.inventory.remove(item_give)
+            self.inventory.append(item_get)
+            friend.inventory.remove(item_get)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, friend):
+        if self.inventory and friend.inventory:
+            self.swap_items(friend, self.inventory[0], friend.inventory[0])
```

Comment:
Nice function reuse.

This method is an interesting look at the balance between code re-use or custom implementations that take into account additional information for better performance. We _could_ choose to reuse our previous code. _Or_ we could specialize our implementation to achieve better performance for this particular scenario. Is there a O(1) solution to this function?